### PR TITLE
Support for Backend Storage Bucket & Examples.

### PR DESCRIPTION
**Pull request by wapfel**
Adds functionality to host static website content behind a HTTP lb using a Cloud Storage Bucket and CDN. Also includes 2 examples, one just using an IP over http and another using a custom domain, https, and ssl.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: autogen/main.tf.tmpl [hunk 1]
```diff
@@ -236,8 +240,27 @@ resource "google_compute_backend_service" "default" {
   }
   {% endif %}
 }
+{% else %}
+resource "google_compute_backend_bucket" "default" {
```
Comment (morgante) [inline autogen/main.tf.tmpl:244]: We should do this as a for_each so multiple buckets can be provided.

File: autogen/main.tf.tmpl [hunk 2]
```diff
@@ -236,8 +240,27 @@ resource "google_compute_backend_service" "default" {
...
+  provider    = google-beta
+  project     = var.project
+  name        = "${var.name}-backend-bucket"
+  description = "connects the GCLB to a backend storage bucket, likely for serving up a statuc website or files"
```
Comment (morgante) [inline autogen/main.tf.tmpl:248]: Descriptions should be capitalized.

File: autogen/variables.tf.tmpl [hunk 3]
```diff
@@ -236,3 +235,52 @@ variable "random_certificate_suffix" {
...
   default     = false
 }
+{% if backend_bucket %}
+variable "bucket_name" {
+  description = "a string value representing the name of the cloud storage bucket for the backend"
+  type        = string
+  default     = null
+}
+
+variable "enable_cdn_for_bucket" {
```
Comment (morgante) [inline autogen/variables.tf.tmpl:245]: I think these CDN options are valid for *all* backends (not just buckets), so we shouldn't gate them here. Maybe just remove them for now so we can discuss in a follow-up PR.

**Review by morgante — CHANGES_REQUESTED** [review]

File: autogen/main.tf.tmpl (hunk 2 above)
Comment (wapfel) [reply autogen/main.tf.tmpl:248]: roger. resolving

File: autogen/main.tf.tmpl (hunk 1 above)
Comment (wapfel) [reply autogen/main.tf.tmpl:244]: I agree, but if a user provides more than 1 bucket, they will then need to provide their own url map. Not sure how best to enforce this or communicate it.

File: autogen/variables.tf.tmpl (hunk 3 above)
Comment (wapfel) [reply autogen/variables.tf.tmpl:245]: it seems to be specific to the Backend_Bucket resource based on the resource variables: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_backend_bucket. I'm open to removing it, but we should discuss first.

**Comment by wapfel** [comment]
closing this pull request and issuing another that addresses all feedback

**Pull request closed without merging**